frontend/src/components/TransactionItem.tsx

```diff
@@ -74,14 +74,22 @@ function TransactionItem({ tx }: Props) {
   const pubkey = tx.metadata?.nostr?.pubkey;
   const npub = pubkey ? safeNpubEncode(pubkey) : undefined;
 
-  const from = tx.metadata?.payer_data?.name
-    ? `from ${tx.metadata.payer_data.name}`
+  const payerName = tx.metadata?.payer_data?.name;
+  const from = payerName
+    ? `from ${payerName}`
     : npub
       ? `zap from ${npub.substring(0, 12)}...`
       : undefined;
 
+  const recipientIdentifier = tx.metadata?.recipient_data?.identifier;
+  const to = recipientIdentifier
+    ? `${tx.state === "failed" ? "sending " : ""}to ${recipientIdentifier}`
+    : undefined;
+
   const eventId = tx.metadata?.nostr?.tags?.find((t) => t[0] === "e")?.[1];
 
+  const description = tx.description || tx.metadata?.comment;
```

Comment:
can we add a comment about how the metadata.comment can be set?

this can be set by the payer for LNURL payments

Reply:
Actually, it seems to not be set. I tried to pay to rolznz@getalby.com with a custom comment and it just set it as the invoice description, and didn't add the comment to the metadata. I think we need to add it to the metadata too

Reply:
Yeah you are correct, we use this comment as description of the invoice :/ Wouldn't it then be redundant in the modal (metadata.comment and description)?

But I think it's good to follow the spec anyways. Should we then change rails app to set the comment to metadata instead of description?

Reply:
No, we can't just consider Alby lightning addresses. Other lightning addresses do not set the description. So I'll re-open the other conversation